modules/database-commons/src/main/java/org/testcontainers/delegate/AbstractDatabaseDelegate.java

```diff
@@ -0,0 +1,61 @@
+package org.testcontainers.delegate;
+
+import java.util.Collection;
+
+/**
+ * @param <CONTAINER>  testcontainers container
+ * @param <CONNECTION> connection to the database
+ * @author Eugeny Karpov
+ */
+public abstract class AbstractDatabaseDelegate<CONTAINER, CONNECTION> implements DatabaseDelegate {
+
+    /**
+     * Testcontainers container
+     */
+    protected CONTAINER container;
+
+    /**
+     * Database connection
+     */
+    private CONNECTION connection;
+
+    public AbstractDatabaseDelegate(CONTAINER container) {
+        this.container = container;
+    }
+
+    /**
+     * Get or create new connection to the database
+     */
+    protected CONNECTION getConnection() {
+        if (connection == null) {
```

Comment:
not thread safe

Reply:
The environment where database delegates are invoked is single threaded, so from my perspective there is no reason to implement here thread safety. Delegate is used to apply a script. No need to even have multiple threads here.